src/db/write.rs

```diff
@@ -338,15 +344,39 @@ impl<'a> Write<'a> {
 
     // Return value is the hash of the new commit.
     #[allow(clippy::too_many_arguments)]
-    pub async fn commit(mut self, head_name: &str) -> Result<String, CommitError> {
+    pub async fn commit(
+        mut self,
+        head_name: &str,
+        generate_diffs: bool,
```

Comment:
I think we should use the name `ChangedKeySet` or `ChangeSet` or something similar to avoid confusion with the term `diff` which we already use elsewhere to mean something different.

Reply:
I agree.